api/src/test/java/com/epam/pipeline/acl/folder/FolderApiServiceTest.java

```diff
@@ -0,0 +1,632 @@
+import com.epam.pipeline.entity.pipeline.Folder;
+import com.epam.pipeline.entity.pipeline.Pipeline;
+import com.epam.pipeline.entity.security.acl.AclClass;
+import com.epam.pipeline.manager.EntityManager;
+import com.epam.pipeline.manager.pipeline.FolderManager;
+import com.epam.pipeline.security.acl.AclPermission;
+import com.epam.pipeline.test.acl.AbstractAclTest;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ALL_PERMISSIONS;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID_2;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID_3;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.READ_PERMISSION;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_STRING;
+import static com.epam.pipeline.test.creator.configuration.ConfigurationCreatorUtils.getRunConfiguration;
+import static com.epam.pipeline.test.creator.datastorage.DatastorageCreatorUtils.getS3bucketDataStorage;
+import static com.epam.pipeline.test.creator.folder.FolderCreatorUtils.getFolder;
+import static com.epam.pipeline.test.creator.folder.FolderCreatorUtils.getFolderWithMetadata;
+import static com.epam.pipeline.test.creator.pipeline.PipelineCreatorUtils.getPipeline;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+public class FolderApiServiceTest extends AbstractAclTest {
+
+    private static final long ID_4 = 4L;
+    private static final long ID_5 = 5L;
+    private static final long ID_6 = 6L;
+    private static final long ID_7 = 7L;
+    private static final long ID_8 = 8L;
+    private static final long ID_9 = 9L;
+    private static final long ID_10 = 10L;
+    private static final long ID_11 = 11L;
+    private static final long ID_12 = 12L;
+    private static final long ID_100 = 100L;
+    private static final long ID_110 = 110L;
+    private static final String FOLDER_MANAGER = "FOLDER_MANAGER";
+    private final Folder folder = getFolder(ID, ID_100, ANOTHER_SIMPLE_USER);
+    final Folder folder10 = getFolder(ID_100, ID_110, ANOTHER_SIMPLE_USER);
+    private final FolderWithMetadata folderWithMetadata = getFolderWithMetadata(ID, ID_110, ANOTHER_SIMPLE_USER);
+    private final Folder emptyChildFolderWithoutPermission = getFolder(ID_3, ID, ANOTHER_SIMPLE_USER);
+    private final Pipeline pipelineRead1 = getPipeline(ID, ANOTHER_SIMPLE_USER);
```

Comment:
As I can see, it would be enough to observe ID's uniqueness within one ACL class. For example, the ID's for `DataStorage` can be `ID`, `ID_2`, `ID_3`, `ID_4`. The way we don't need to create to many ID variables. Could you please check and correct if my observation is correct?